crates/driver/src/domain/competition/mod.rs

```diff
@@ -190,16 +190,20 @@ impl Competition {
             let simulate_on_new_blocks = async move {
                 let mut stream =
                     ethrpc::current_block::into_stream(self.eth.current_block().clone());
+                let mut simulated_once = false;
                 while let Some(block) = stream.next().await {
                     if let Err(err) = self.simulate_settlement(&settlement).await {
                         observe::winner_voided(block, &err);
                         *score_ref = None;
                         *self.settlement.lock().unwrap() = None;
                         if let Some(id) = settlement.notify_id() {
-                            notify::simulation_failed(&self.solver, auction.id(), id, &err);
+                            notify::simulation_failed(&self.solver, auction.id(), id, &err, simulated_once);
```

Comment:
I guess I am misinterpreting the code then. So if the condition on (new) line 195 is true in the first iteration of the loop, then it would return that it simulated successfully at least once, where in fact it hasn't, right? What am I missing?

Reply:
The first ever simulation happens when the solution is encoded, so the successful simulation is the prerequisite for a solution to be encoded.

This whole code block is not even executed if the first simulation failed. So, if the simulation fails here, it means that is the second, third .... attempt.

Reply:
Got you, thanks!